tensorboard/plugins/text/summary.py

```diff
@@ -0,0 +1,113 @@
+  """Summarizes textual data.
+
+  Text data summarized via this plugin will be visible in the Text Dashboard
+  in TensorBoard. The standard TensorBoard Text Dashboard will render markdown
+  in the strings, and will automatically organize 1d and 2d tensors into tables.
+  If a tensor with more than 2 dimensions is provided, a 2d subarray will be
+  displayed along with a warning message. (Note that this behavior is not
+  intrinsic to the text summary api, but rather to the default TensorBoard text
+  plugin.)
+
+  Args:
+    name: A name for the generated node. Will also serve as a series name in
+      TensorBoard.
+    data: a string-type Tensor to summarize. The text should be UTF-encoded.
+    display_name: Optional name for this summary in TensorBoard, as a
+      constant `str`. Defaults to `name`.
+    description: Optional long-form description for this summary, as a
+      constant `str`. Markdown is supported. Defaults to empty.
+    collections: Optional list of ops.GraphKeys.  The collections to add the
+      summary to.  Defaults to [_ops.GraphKeys.SUMMARIES]
+
+  Returns:
+    A TensorSummary op that is configured so that TensorBoard will recognize
+    that it contains textual data. The TensorSummary is a scalar `Tensor` of
+    type `string` which contains `Summary` protobufs.
+
+  Raises:
+    ValueError: If tensor has the wrong type.
+  """
+  if display_name is None:
+    display_name = name
+  summary_metadata = metadata.create_summary_metadata(
+      display_name=display_name, description=description)
+  with tf.name_scope(name):
+    with tf.control_dependencies([tf.assert_type(data, tf.string)]):
+      return tf.summary.tensor_summary(name='text_summary',
+                                       tensor=data,
+                                       collections=collections,
+                                       summary_metadata=summary_metadata)
+
+
+def pb(name, data, display_name=None, description=None):
+  """Create a text summary protobuf.
+
+  Arguments:
+    name: A name for the generated node. Will also serve as a series name in
+      TensorBoard.
+    data: A Python bytestring (of type bytes), or Unicode string, or numpy array
+      of one of these types.
+    display_name: Optional name for this summary in TensorBoard, as a
+      `str`. Defaults to `name`.
+    description: Optional long-form description for this summary, as a
+      `str`. Markdown is supported. Defaults to empty.
+
+  Raises:
+    ValueError: If the data is of the wrong type.
+
+  Returns:
+    A `tf.Summary` protobuf object.
+  """
+  if isinstance(data, (six.binary_type, six.string_types)):
+    if isinstance(data, six.text_type):
+      # This is a unicode string.
+      data = tf.compat.as_bytes(data)
+    data = np.array(data)
+  if data.dtype.kind not in ('S', 'U'):
+    raise ValueError(
+        'Type %r is not supported. Only strings are.' % data.dtype.name)
+  tensor = tf.make_tensor_proto(data, dtype=tf.string)
```

Comment:
Can't this whole block be replaced with
```py
tensor = tf.make_tensor_proto(data, dtype=tf.string)
```
? This will handle both numpy arrays and scalar values, both byte or text strings, in both Python 2 and 3. It will throw a `TypeError` if the type does not match; feel free to catch that and rethrow a `ValueError` with the same message.

Reply:
Yes - what if we just relied on the `TypeError` raised by `make_tensor_proto `?

Reply:
That's fine with me; it does make sense. I suggested keeping it a `ValueError` for consistency (all plugins currently throw `ValueError`s for all `pb` errors).

Reply:
Actually, yes, +1 to consistency. Changed to catching the TypeError and raising a ValueError.